src/tqdm_publisher/_handler.py

```diff
@@ -1,27 +1,46 @@
 import queue
+from typing import Any, Dict, Iterable, List
 
 from ._subscriber import TQDMProgressSubscriber
 
 
 class TQDMProgressHandler:
     def __init__(self):
-        self.listeners = []
-
-    def listen(self):
-        q = queue.Queue(maxsize=25)
-        self.listeners.append(q)
-        return q
-
-    def create(self, iterable, additional_metadata: dict = dict(), **tqdm_kwargs):
-        return TQDMProgressSubscriber(
-            iterable,
-            lambda progress_update: self._announce(dict(**progress_update, **additional_metadata)),
-            **tqdm_kwargs,
-        )
-
-    def _announce(self, msg):
-        for i in reversed(range(len(self.listeners))):
-            try:
-                self.listeners[i].put_nowait(msg)
-            except queue.Full:
-                del self.listeners[i]
+        self.listeners: List[queue.Queue] = []
+
+    def listen(self) -> queue.Queue:
+        new_queue = queue.Queue(maxsize=25)
+        self.listeners.append(new_queue)
+        return new_queue
+
+    def create_progress_subscriber(
+        self, iterable: Iterable[Any], additional_metadata: dict = dict(), **tqdm_kwargs
+    ) -> TQDMProgressSubscriber:
+
+        def on_progress_update(progress_update: dict):
+            """
+            This is the injection called on every update of the progress bar.
+
+            It triggers the announcement event over all listeners on each update of the progress bar.
+
+            It must be defined inside this local scope to communicate the `additional_metadata` from the level above
+            without including it in the method signature.
+            """
+            self._announce(message=dict(**progress_update, **additional_metadata))
+
+        return TQDMProgressSubscriber(iterable=iterable, on_progress_update=on_progress_update, **tqdm_kwargs)
+
+    def _announce(self, message: Dict[Any, Any]):
+        """
+        Announce a message to all listeners.
+
+        @garrett - can you describe the expected structure of this message?
```

Comment:
@garrettmflynn attn

Reply:
Updated. Let me know if this is sufficient.

```
    def _announce(self, message: Dict[Any, Any]):
        """
        Announce a message to all listeners.

        This message can be any dictionary. But, when used internally, is 
        expected to contain the progress_bar_id and format_dict of the TQDMProgressSubscriber class,
        as well as any additional metadata supplied by the create_progress_subscriber method.

        """
```

Reply:
Cool thanks!